test/main.py

```diff
@@ -22,86 +23,61 @@ def generate_indent(number):
 
 class CiscoConfig:
     nodes = []
+    interfaces = []
+    root = None
+    indent = None
+    parent = None
+    config = None
+    i = None
 
     def __init__(self, file):
         self.file = file
         self.fd = open(file, "r")
-        root = Node("root")
-        config = self.create_entry_list(self.fd.readlines())
-
-        i = iter(config)
-        indent = 0
-        parent = root
+        self.config = self.create_entry_list(self.fd.readlines())
+        self.indent = 0
+        self.i = iter(self.config)
+        self.root = Node("root")
+        self.parent = self.root
 
+    def create_nodes(self, parent, i, indent):
         while True:
             try:
                 line = next(i)
                 if indent == (len(line) - len(line.lstrip('\t')) - 1):
-                    parent = parent.childs[-1]
-                    node = Node(line.lstrip('\t'))
-                    node.parent = parent
-                    indent = indent +1
+                    parent = parent.children[-1]
+                    Node(line.lstrip('\t'), parent=parent)
```

Comment:
Spróbowałbym podziedzicic po klasie Node i stworzył klasę CiscoNode i dodał pole property(enuma). Do tego enuma, który bym trzymał typy nodow w zakresie cisco, czyli Miscellonues, Ethernet Interface itd. (ograniczylbym się do tych dwóch) stworzył bym metodę która w trakcie parsowania byłaby w stanie ustalić property i już je ustawiać. Może się to przydać w dalszej części programu.

Reply:
Do tych node'ów można dodawać dowolne atrybuty za pomocą których mozna potem wyszukać node w drzewie

Reply:
No to enuma tylko wystarczy w takim razie tam wrzucić